diff --git a/Framework/Kernel/inc/MantidKernel/ListValidator.h b/Framework/Kernel/inc/MantidKernel/ListValidator.h
index 8ba6159b59a..bdb96f6c967 100644
--- a/Framework/Kernel/inc/MantidKernel/ListValidator.h
+++ b/Framework/Kernel/inc/MantidKernel/ListValidator.h
@@ -17,6 +17,7 @@
 #include <set>
 #include <unordered_set>
 #include <vector>
+#include <algorithm>
 
 namespace Mantid {
 namespace Kernel {
diff --git a/Framework/Kernel/inc/MantidKernel/Strings.h b/Framework/Kernel/inc/MantidKernel/Strings.h
index 90f7d8c5859..bfc2a338d8c 100644
--- a/Framework/Kernel/inc/MantidKernel/Strings.h
+++ b/Framework/Kernel/inc/MantidKernel/Strings.h
@@ -23,6 +23,7 @@
 #include <sstream>
 #include <string>
 #include <vector>
+#include <algorithm>
 
 namespace Mantid {
 namespace Kernel {
diff --git a/Framework/Kernel/inc/MantidKernel/System.h b/Framework/Kernel/inc/MantidKernel/System.h
index 43c77bc0614..bf3f94fe811 100644
--- a/Framework/Kernel/inc/MantidKernel/System.h
+++ b/Framework/Kernel/inc/MantidKernel/System.h
@@ -24,6 +24,7 @@
 /**
  * Definitions of the DLLImport compiler directives for MSVC
  */
+#ifndef DLLExport
 #ifdef _WIN32
 // Export/Import declarations
 #define DLLExport __declspec(dllexport)
@@ -38,6 +39,7 @@
 #define DLLImport
 #define EXTERN_IMPORT
 #endif
+#endif
 
 /**
  * Function arguments are sometimes unused in certain implmentations
diff --git a/Framework/Kernel/src/DllOpen.cpp b/Framework/Kernel/src/DllOpen.cpp
index cda55fc6ddd..cc663adafee 100644
--- a/Framework/Kernel/src/DllOpen.cpp
+++ b/Framework/Kernel/src/DllOpen.cpp
@@ -72,7 +72,7 @@ void DllOpen::closeDll(void *handle) { FreeLibrary((HINSTANCE)handle); }
 #else
 
 const std::string LIB_PREFIX = "lib";
-#ifdef __linux__
+#if defined __linux__ || defined __EMSCRIPTEN__
 const std::string LIB_SUFFIX = ".so";
 #elif defined __APPLE__
 const std::string LIB_SUFFIX = ".dylib";
diff --git a/Framework/LegacyNexus/inc/MantidLegacyNexus/napiconfig.h b/Framework/LegacyNexus/inc/MantidLegacyNexus/napiconfig.h
index 9b6c505bf7d..3e6beac9197 100644
--- a/Framework/LegacyNexus/inc/MantidLegacyNexus/napiconfig.h
+++ b/Framework/LegacyNexus/inc/MantidLegacyNexus/napiconfig.h
@@ -4,8 +4,8 @@
 #include <windows.h>
 #endif /* _WIN32 */
 
-#define HAVE_HDF4
-#define WITH_HDF4
+//#define HAVE_HDF4
+//#define WITH_HDF4
 
 #define HAVE_HDF5
 #define WITH_HDF5
diff --git a/Framework/DataObjects/inc/MantidDataObjects/EventList.h b/Framework/DataObjects/inc/MantidDataObjects/EventList.h
index 20c9c9a6827..71f6189c0f4 100644
--- a/Framework/DataObjects/inc/MantidDataObjects/EventList.h
+++ b/Framework/DataObjects/inc/MantidDataObjects/EventList.h
@@ -14,6 +14,7 @@
 
 #include <iosfwd>
 #include <vector>
+#include <optional>
 
 namespace Mantid {
 namespace Types {
diff --git a/Framework/PythonInterface/mantid/dataobjects/src/Exports/WorkspaceValidators.cpp b/Framework/PythonInterface/mantid/dataobjects/src/Exports/WorkspaceValidators.cpp
index 193daa3bb7d..0ac26cca2ba 100644
--- a/Framework/PythonInterface/mantid/dataobjects/src/Exports/WorkspaceValidators.cpp
+++ b/Framework/PythonInterface/mantid/dataobjects/src/Exports/WorkspaceValidators.cpp
@@ -45,7 +45,7 @@ void export_TableWorkspaceValidator() {
   class_<ValidatorType, bases<TableWorkspaceValidator>, boost::noncopyable>(                                           \
       #ValidatorType, init<ArgType>(arg(ArgName) = DefaultValue, DocString));
 
-void export_WorkspaceValidators() {
+void export_WorkspaceValidators_DataObjects() {
   using namespace Mantid::DataObjects;
 
   EXPORT_WKSP_VALIDATOR_NO_ARG(TableWorkspaceNotEmptyValidator, "Checks that the workspace is not empty");
diff --git a/Framework/Kernel/inc/MantidKernel/Math/Distributions/ChebyshevPolynomial.h b/Framework/Kernel/inc/MantidKernel/Math/Distributions/ChebyshevPolynomial.h
index 24fb94a8550..6781b4ff5f3 100644
--- a/Framework/Kernel/inc/MantidKernel/Math/Distributions/ChebyshevPolynomial.h
+++ b/Framework/Kernel/inc/MantidKernel/Math/Distributions/ChebyshevPolynomial.h
@@ -10,6 +10,7 @@
 // Includes
 //-----------------------------------------------------------------------------
 #include "MantidKernel/DllConfig.h"
+#include <iterator>
 
 namespace Mantid {
 namespace Kernel {
