diff --git a/Framework/PythonInterface/core/src/ErrorHandling.cpp b/Framework/PythonInterface/core/src/ErrorHandling.cpp
index a2f61284629..6a61faea95e 100644
--- a/Framework/PythonInterface/core/src/ErrorHandling.cpp
+++ b/Framework/PythonInterface/core/src/ErrorHandling.cpp
@@ -39,7 +39,7 @@ void tracebackToStream(std::ostream &msg, PyTracebackObject *traceback, bool roo
     msg << "caused by";

   msg << " line " << traceback->tb_lineno << " in \'"
-      << extract<const char *>(traceback->tb_frame->f_code->co_filename)() << "\'";
+      << extract<const char *>(PyFrame_GetCode(traceback->tb_frame)->co_filename)() << "\'";
   tracebackToStream(msg, traceback->tb_next, false);
 }

diff --git a/Framework/PythonInterface/mantid/kernel/funcinspect.py b/Framework/PythonInterface/mantid/kernel/funcinspect.py
index 9a424746b8a..dcb0baf7948 100644
--- a/Framework/PythonInterface/mantid/kernel/funcinspect.py
+++ b/Framework/PythonInterface/mantid/kernel/funcinspect.py
@@ -210,6 +210,8 @@ OPERATOR_NAMES = {
     "LIST_EXTEND",
     "SET_UPDATE",
     "BUILD_CONST_KEY_MAP",
+    "CALL",
+    "BINARY_OP",
 }


