diff --git a/Framework/PythonInterface/mantid/kernel/funcinspect.py b/Framework/PythonInterface/mantid/kernel/funcinspect.py
index 5ec0554b529..a53efa14374 100644
--- a/Framework/PythonInterface/mantid/kernel/funcinspect.py
+++ b/Framework/PythonInterface/mantid/kernel/funcinspect.py
@@ -182,6 +182,8 @@ OPERATOR_NAMES = {
     "LIST_EXTEND",
     "SET_UPDATE",
     "BUILD_CONST_KEY_MAP",
+    "CALL",
+    "BINARY_OP",
 }
 
 
@@ -234,7 +236,10 @@ def process_frame(frame):
         last_i = [offset for offset in call_function_locs.keys() if offset > last_i][0]
 
     output_var_names = []
-    last_func_offset = call_function_locs[last_i][0]
+    try:
+        last_func_offset = call_function_locs[last_i][0]
+    except KeyError: # A weird offset which sometimes happens in Python 3.12
+        last_func_offset = call_function_locs[last_i - 6][0]
     # On Windows since migrating to Python 3.10, the last instruction index appears
     # to be one step behind where it should be. We think it's related to the comment
     # here:
